src/main/kotlin/org/rust/lang/core/lexer/RustStringLiteralLexer.kt

```diff
@@ -0,0 +1,129 @@
+package org.rust.lang.core.lexer
+
+import com.intellij.openapi.util.text.StringUtil
+import com.intellij.psi.StringEscapesTokenTypes.*
+import com.intellij.psi.tree.IElementType
+
+const val BYTE_ESCAPE_LENGTH = "\\x00".length
+const val UNICODE_ESCAPE_MIN_LENGTH = "\\u{0}".length
+const val UNICODE_ESCAPE_MAX_LENGTH = "\\u{000000}".length
+
+/**
+ * Performs lexical analysis of Rust byte/char/string/byte string literals using Rust character escaping rules.
+ */
+public class RustStringLiteralLexer(val defaultToken: IElementType,
+                                    val byte: Boolean = false,
```

Comment:
Hm, and what might be the reason to turn it off? I'd rather remove it for know if there is no immediate need in this: